src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -101,13 +97,15 @@ public boolean Disconnect(String peerId){
 // we only cut off our outgoing connections, incoming ones are cut off from the other end.
 // if we want to cut off whole communications, we'll do Stop
         if(mBtToRequestSocket != null) {
```

Comment:
@DrJukka I don't believe this is safe. You can't synchronize on an object that can become null, I believe it will cause an exception. A much simpler approach is to just make a local copy of mBtToRequestSocket and then test if the local copy is null and issue methods based on that. This assumes, of course that all of mBtToRequestSocket's methods are thread safe but of course they should be given our asynchronous environment.